module-system/module-implementations/sov-evm/src/evm/conversions.rs

```diff
@@ -47,65 +53,57 @@ impl From<BlockEnv> for ReVmBlockEnv {
-    fn from(tx: EvmTransaction) -> Self {
-        let to = match tx.to {
-            Some(addr) => TransactTo::Call(B160::from_slice(&addr)),
+        let to = match tx.to() {
+            Some(addr) => TransactTo::Call(addr),
             None => TransactTo::Create(CreateScheme::Create),
         };
 
-        let access_list = tx
-            .access_lists
-            .into_iter()
-            .map(|item| item.into())
-            .collect();
-
         Self {
-            caller: B160::from_slice(&tx.sender),
-            data: Bytes::from(tx.data),
-            gas_limit: tx.gas_limit,
-            gas_price: U256::from(tx.gas_price),
-            gas_priority_fee: Some(U256::from(tx.max_priority_fee_per_gas)),
+            caller: tx.signer(),
+            gas_limit: tx.gas_limit(),
+            gas_price: U256::from(tx.effective_gas_price(None)),
+            gas_priority_fee: tx.max_priority_fee_per_gas().map(U256::from),
             transact_to: to,
-            value: U256::from(tx.value),
-            nonce: Some(tx.nonce),
-            chain_id: Some(tx.chain_id),
-            access_list,
+            value: U256::from(tx.value()),
+            data: Bytes::from(tx.input().to_vec()),
+            chain_id: tx.chain_id(),
+            nonce: Some(tx.nonce()),
+            // TODO handle access list
+            access_list: vec![],
         }
     }
 }
 
-impl From<EvmTransaction> for Transaction {
-    fn from(evm_tx: EvmTransaction) -> Self {
-        Self {
-            hash: evm_tx.hash.into(),
-            nonce: evm_tx.nonce.into(),
-            from: evm_tx.sender.into(),
-            to: evm_tx.to.map(|addr| addr.into()),
-            value: evm_tx.value.into(),
-            // https://github.com/foundry-rs/foundry/blob/master/anvil/core/src/eth/transaction/mod.rs#L1251
-            gas_price: Some(evm_tx.max_fee_per_gas.into()),
-            input: evm_tx.data.into(),
-            v: (evm_tx.odd_y_parity as u8).into(),
-            r: evm_tx.r.into(),
-            s: evm_tx.s.into(),
-            transaction_type: Some(2.into()),
+impl TryFrom<RawEvmTransaction> for Transaction {
+    type Error = RawEvmTxConversionError;
+    fn try_from(evm_tx: RawEvmTransaction) -> Result<Self, Self::Error> {
+        let tx: EvmTransactionSignedEcRecovered = evm_tx.try_into()?;
+        let tx: &RethTransactionSignedEcRecovered = tx.as_ref();
+
+        Ok(Self {
+            hash: tx.hash().into(),
+            nonce: tx.nonce().into(),
+
+            from: tx.signer().into(),
+            to: tx.to().map(|addr| addr.into()),
+            value: tx.value().into(),
+            gas_price: Some(tx.effective_gas_price(None).into()),
+
+            input: EthBytes::from(tx.input().to_vec()),
+            // TODO
+            v: tx.signature().v(None).into(),
+            r: tx.signature().r.into(),
+            s: tx.signature().s.into(),
+            transaction_type: Some(1u64.into()),
+            // // TODO handle access list
```

Comment:
You need to match `TxEip2930` or `TxEip1559` to get access list

Reply:
This will be done in #503